ui-tests/cypress/integration/content-types.spec.js

```diff
@@ -2,236 +2,290 @@ import {
+
+    contentTypeCode = generateRandomContentTypeCode();
+    contentTypeName = generateRandomId();
   });
 
   afterEach(() => {
-    cy.appBuilderLogout();
+    cy.kcLogout();
   });
 
-  const contentTypeCode = 'BNR';
-  const TYPE_LIST = 'List';
-  const TYPE_MONOLIST = 'Monolist';
-  const TYPE_COMPOSITE = 'Composite';
-
-  const attributeCompositeTest = [
-    {
-      type: 'Hypertext',
-      codeValue: 'httext',
-      nameEnValue: 'Le Hyper Text',
-    }, {
-      type: 'Link',
-      codeValue: 'myLink',
-      nameEnValue: 'My Link',
-    }, {
-      type: 'Timestamp',
-      codeValue: 'currStamp',
-      nameEnValue: 'Curr Stamp',
-    },
-  ];
-
-  describe('Content Type Attribute - List', () => {
-    describe('nested attribute types that are not allowed in List attribute', () => {
-      it('attribute type selection should not contain Text, Longtext, Hypertext, Image, Attach, Link', () => {
-        cy.openContentTypeFormWith(contentTypeCode);
-        cy.addNewContentTypeAttribute(contentTypeCode, TYPE_LIST);
-        cy.getByName(TEST_ID_CONTENTTYPE_FORM.ATTRIBUTE_TYPE_DROPDOWN).should('not.contain', 'Text');
-        cy.getByName(TEST_ID_CONTENTTYPE_FORM.ATTRIBUTE_TYPE_DROPDOWN).should('not.contain', 'Longtext');
-        cy.getByName(TEST_ID_CONTENTTYPE_FORM.ATTRIBUTE_TYPE_DROPDOWN).should('not.contain', 'Hypertext');
-        cy.getByName(TEST_ID_CONTENTTYPE_FORM.ATTRIBUTE_TYPE_DROPDOWN).should('not.contain', 'Image');
-        cy.getByName(TEST_ID_CONTENTTYPE_FORM.ATTRIBUTE_TYPE_DROPDOWN).should('not.contain', 'Attach');
-        cy.getByName(TEST_ID_CONTENTTYPE_FORM.ATTRIBUTE_TYPE_DROPDOWN).should('not.contain', 'Link');
-      });
-    });
+  const addContentType = (code, name) => {
+    cy.log(`Add content type with code ${code}`);
+    currentPage = currentPage.getContent().addContentType();
+    currentPage = currentPage.getContent().addAndSaveContentType(code, name);
+  };
+
+  it('should have the functionality to add a new content type', () => {
+    addContentType(contentTypeCode, contentTypeName);
+    currentPage.getContent().getCodeInput().should('have.value', contentTypeCode).and('be.disabled');
+    currentPage.getContent().getNameInput().should('have.value', contentTypeName);
+    currentPage = currentPage.getContent().save();
+    currentPage.getContent().getTableCell(contentTypeCode).should('be.visible');
+    currentPage.getContent().getTableCell(contentTypeName).should('be.visible');
+    cy.log(`Delete content type with code ${contentTypeCode}`);
+    currentPage.getContent().deleteContentType(contentTypeCode);
+  });
 
-    describe('examples of nested attribute types that are allowed in List attribute', () => {
-      beforeEach(() => {
-        cy.openContentTypeFormWith(contentTypeCode);
-        cy.addNewContentTypeAttribute(contentTypeCode, TYPE_LIST);
-      });
+  it('should have the functionality to edit a content type', () => {
+    addContentType(contentTypeCode, contentTypeName);
```

Comment:
You should use the API to setup the status needed for the test; in this way we'll have more confidence asserting that the functionality under test is not working if the test fails